File: src/Marvin.Products.Management/Modification/Model/ProductModel.cs
// Copyright (c) 2020, Phoenix Contact GmbH & Co. KG
// Licensed under the Apache License, Version 2.0

using System.Runtime.Serialization;
using Marvin.AbstractionLayer;
using Marvin.Serialization;

namespace Marvin.Products.Management.Modification
{
    [DataContract(IsReference = true)]
    internal class ProductModel
    {
        [DataMember]
        public long Id { get; set; }

        [DataMember]
        public string Name { get; set; }

        [DataMember]
        public ProductTypeState State { get; set; }

        [DataMember]
        public string Identifier { get; set; }

        [DataMember]
        public short Revision { get; set; }

        [DataMember]
        public string Type { get; set; }

        [DataMember]
        public Entry Properties { get; set; }

        [DataMember]
        public ProductFile[] Files { get; set; }

        [DataMember]
        public PartConnector[] Parts { get; set; }

        [DataMember]
        public RecipeModel[] Recipes { get; set; }
    }
}

File: src/Moryx.Products.Web/src/app/api/models/product-model.ts
/* tslint:disable */
/* eslint-disable */
import { Entry } from '@moryx/ngx-web-framework/entry-editor';
import { PartConnector } from './part-connector';
import { ProductState } from './product-state';
import { RecipeModel } from './recipe-model';
export interface ProductModel {
  id?: number;
  identifier?: null | string;
  name?: null | string;
  parts?: null | Array<PartConnector>;
  properties?: Entry;
  recipes?: null | Array<RecipeModel>;
  revision?: number;
  state?: ProductState;
  type?: null | string;
}
